### ISSUE TITLE
`no-useless-constructor` / `no-empty-function` with mixed JS & TS codebase

### ISSUE THREAD

**Issue by scottohara** [nzakas/eslint-plugin-typescript#101]
(This issue was originally raised in https://github.com/eslint/typescript-eslint-parser/issues/418; and it was suggested that it would be more appropriately logged here, as it is not something that the parser can handle)

**What version of TypeScript are you using?**
2.6.2

**What version of `typescript-eslint-parser` are you using?**
9.0.0

**What code were you trying to parse?**

```typescript
export default class Foo {
  constructor(private name: string) {}

  get greeting() : string {
    return `Hello ${this.name}`;
  }
}
```

**What did you expect to happen?**
No errors/warnings about empty constructor.

**What happened?**

```
  2:2   error  Useless constructor           no-useless-constructor
  2:36  error  Unexpected empty constructor  no-empty-function
```

For now, I have turned off these two rules in my `.ts` projects; but for projects that mix `*.ts` and `*.js` code, it would be nice to be able to have these rules enabled to catch any useless constructors/empty functions in `*.js` code without it also warning about valid `*.ts` constructors.

**Comment by macklinu** [comment]
I'm curious what the proper solution is here. Should these rules be implemented in this plugin to support TypeScript's constructor syntax, since `constructor(private name: string) {}` is the equivalent of:

```ts
class Foo {
  private name: string
  constructor(name: string) {
    this.name = name
  }
}
```

I'd like to help out, but I'm not sure I understand what the next steps should be.

**Comment by corbinu** [comment]
@macklinu

Yes the eslint rule should be forked into this plugin and fixed to understand TypeScript just like https://github.com/nzakas/eslint-plugin-typescript/blob/master/docs/rules/no-unused-vars.md

**Comment by macklinu** [comment]
Cool, thanks for that info, @corbinu!
